### PR TITLE
Rewrite .js files into .ts files

### PR DESCRIPTION
**Pull request by caphosra**
As described in the title, I rewrote backend files in Javascript into those in Typescript. Since the strict option is currently disabled, the migration hasn't finished yet.

### CODE REVIEW

**Review by n4o847 — COMMENTED** [review]
Great work!
The code is now typed, which makes it easier for everyone to develop.

Here are my personal concerns:
- Have you considered something like `ts-node`, a tool that can run TypeScript files directly without build steps? (It can be used in combination with nodemon)
- The code seems to be formatted with `prettier-eslint` (very nice), but `eslint` itself fails. (Maybe another PR?)

File: bin/writeup-gen.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-const langs = require('../data/langs.json');
-const {stripIndent} = require('common-tags');
+import {stripIndent} from 'common-tags';
+import mongoose from 'mongoose';
+import isValidUTF8 from 'utf-8-validate';
+import langs from '../data/langs.json';
+import Contest from '../models/Contest';
+import Language from '../models/Language';
+import _ from '../models/Submission';
+import __ from '../models/User';
```
Comment (n4o847) [inline bin/writeup-gen.ts]: [nits] How about this?

```suggestion
import '../models/Submission';
import '../models/User';
```

File: bin/disasm.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-const Language = require('../models/Language');
-const Submission = require('../models/Submission');
-
-require('../models/User');
+import mongoose from 'mongoose';
+import langs from '../data/langs.json';
+import docker from '../engines/docker';
+import Language from '../models/Language';
+import Submission from '../models/Submission';
+import _ from '../models/User';
```
Comment (n4o847) [inline bin/disasm.ts]: [nits]

```suggestion
import '../models/User';
```

File: bin/rejudge.ts [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
-
-require('../models/User');
+import assert from 'assert';
+import mongoose from 'mongoose';
+import languagesData from '../data/languages';
+import {validate} from '../lib/validation';
+import Contest from '../models/Contest';
+import Language from '../models/Language';
+import Submission from '../models/Submission';
+import _ from '../models/User';
```
Comment (n4o847) [inline bin/rejudge.ts]: [nits]

```suggestion
import '../models/User';
```

**Comment by caphosra** [comment]
Thank you for your review!
I removed unused variables named `_` as you suggested. But migrating to `ts-node` and a proper setup for `eslint` would be in another PR.

**Pull request merged**